src/dsp_tools/utils/connection_live.py

```diff
@@ -327,15 +340,17 @@ def put(
         if not route.startswith("/"):
             route = f"/{route}"
         url = self.server + route
+        if not headers:
+            headers = {}
```

Comment:
this actually makes me think: should headers be optional and the default be none? or should it simply default to `{}`?

Reply:
If you mean give an empty dictionary as the default parameter that is unfortunately not a good idea. 

https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
then `None` it is :)  
I keep forgetting what a weird language Python is...

Reply:
It is a pity, I was annoyed every time I wrote those two lines...